### PR TITLE
Add LANTERN 5-fold deep-learning plaque segmentation

### PR DESCRIPTION
**Pull request by Arshya-Guru**
Adds an opt-in alternative to the GMM/N4 plaque segmentation

Leaves in the old GMM based method as the default, but using the `--seg_method lantern` will use the ki3 trained amyloid beta plaque detection model (version 2). For now, `--seg_method lantern` will only work on the amyloid beta channel and other non-vessel channels should revert to the default seg_method.

### CODE REVIEW

**Comment by akhanf** [comment]
Sorry for the delay, finally having a chance to look at the outputs from the few test subjects you ran - results look great!
I'll try to have this merged in soon, there is another dataset I'd like to run through in the short-term (mouse_app_mapt_apoe - old data) to see if we are able to effectively deal with the false positive staining there - trying to wrap up a paper very soon on that!

File: spimquant/workflow/scripts/lantern_plaques.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+        tensor = torch.from_numpy(arr).to(device)
+        tensor = normalize_tiles(tensor)[:, None]
+        batch_votes = torch.zeros(
+            (len(batch), tile, tile, tile), dtype=torch.uint8, device=device
+        )
+        with (
+            torch.no_grad(),
+            torch.autocast(
+                device.type, dtype=torch.bfloat16, enabled=device.type == "cuda"
+            ),
```
[the remark this answers is not in the document]
Comment (akhanf) [reply spimquant/workflow/scripts/lantern_plaques.py]: seems like a harmless addition, @Arshya-Guru ok to commit this?

File: spimquant/config/snakebids.yml [hunk 2]
```diff
@@ -447,6 +459,12 @@ templates:
 
 models:
   vesselfm: https://huggingface.co/bwittmann/vesselFM/resolve/main/vesselFM_base.pt
+  lantern_abeta:
+    fold0: https://huggingface.co/apooladi/lantern-ki3-abeta/resolve/main/fold0/seg_model.pt
+    fold1: https://huggingface.co/apooladi/lantern-ki3-abeta/resolve/main/fold1/seg_model.pt
+    fold2: https://huggingface.co/apooladi/lantern-ki3-abeta/resolve/main/fold2/seg_model.pt
+    fold3: https://huggingface.co/apooladi/lantern-ki3-abeta/resolve/main/fold3/seg_model.pt
+    fold4: https://huggingface.co/apooladi/lantern-ki3-abeta/resolve/main/fold4/seg_model.pt
```
[the remark this answers is not in the document]
Comment (akhanf) [reply spimquant/config/snakebids.yml]: I guess the suggestion here would be to use e.g. `b0f6357` instead of `main`? seems reasonable to ensure reproducibility, unless @Arshya-Guru you are able to tag in huggingface?

[the remark this answers is not in the document]
Comment (akhanf) [reply spimquant/config/snakebids.yml]: btw when we do a refactor of spimquant cli in the future maybe could have the model version as an option (which would just set that tag)

File: spimquant/workflow/scripts/lantern_plaques.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Arshya-Guru) [reply spimquant/workflow/scripts/lantern_plaques.py]: Checked this against the env from the pixi lock (pytorch-gpu 2.7.1): the described error corresponds to older generations. Since torch2.4, the autocast gaurd calls `is_bf16_supported()` with emulation included, so autocast enters fine on those cards regardless.

File: spimquant/config/snakebids.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Arshya-Guru) [reply spimquant/config/snakebids.yml]: Yep good suggestions, and I was also thinking the same for the CLI once the next model is trained. Created v2 pointing at the current commit and updating the fold URLs to `/resolve/v2`. 
```suggestion
    fold0: https://huggingface.co/apooladi/lantern-ki3-abeta/resolve/v2/fold0/seg_model.pt
    fold1: https://huggingface.co/apooladi/lantern-ki3-abeta/resolve/v2/fold1/seg_model.pt
    fold2: https://huggingface.co/apooladi/lantern-ki3-abeta/resolve/v2/fold2/seg_model.pt
    fold3: https://huggingface.co/apooladi/lantern-ki3-abeta/resolve/v2/fold3/seg_model.pt
    fold4: https://huggingface.co/apooladi/lantern-ki3-abeta/resolve/v2/fold4/seg_model.pt
```

File: spimquant/workflow/scripts/binarize_lantern_plaques.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    for axis, (have, want) in enumerate(zip(arr.shape, target_shape)):
+        factor = max(1, int(round(want / have)))
+        if factor > 1:
+            out = da.repeat(out, factor, axis=axis)
+
+    out = out[tuple(slice(0, min(h, w)) for h, w in zip(out.shape, target_shape))]
+    pad = [(0, w - h) for h, w in zip(out.shape, target_shape)]
+    if any(after for _, after in pad):
+        out = da.pad(out, pad, mode="edge")
+    return out
```
[the remark this answers is not in the document]
Comment (Arshya-Guru) [reply spimquant/workflow/scripts/binarize_lantern_plaques.py]: This is correct, I reproduced it as well. `plaque-level` is 1 and `--segmentation-_level` is user facing so `--segmentation_level 2` does corner-crop. 

However, I don't think this deserves a code patch right now, because the level ordering it guards has no use case for LANTERN specifically: inference is pinned at plaque_level 1 so unlike the default segmentation methods running lantern with a segmentation level of 2 or 3 is not less computationally cheaper. Copilot's suggested resize wouldnt be the best fit here either as nearest-downsampling a binary plaque mask silently drops plaques smaller than the sampling stride, and the principled coarse representation already exists via the mean-pooled fieldfrac.

We could deal with the error of `--seg_method lantern` with `--segmentation_level > plaque_level` later in the CLI refactor. but leaving this open for you @akhanf

[the remark this answers is not in the document]
Comment (akhanf) [reply spimquant/workflow/scripts/binarize_lantern_plaques.py]: ok I'm fine to just go with what we have here and not make any changes
